open_spiel/games/battleship.cc

```diff
@@ -0,0 +1,849 @@
+// Copyright 2020 DeepMind Technologies Ltd. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.#include "open_spiel/games/battleship.h"
+
+#include "open_spiel/games/battleship.h"
+
+#include "open_spiel/abseil-cpp/absl/strings/ascii.h"
+#include "open_spiel/abseil-cpp/absl/strings/numbers.h"
+#include "open_spiel/abseil-cpp/absl/strings/str_split.h"
+#include "open_spiel/abseil-cpp/absl/strings/string_view.h"
+#include "open_spiel/abseil-cpp/absl/strings/strip.h"
+
+namespace open_spiel {
+namespace battleship {
+
+const double kFloatTolerance = 1e-9;
+
+const Player Player1 = Player{0};
```

Comment:
I think this unnecessarily complicates the code. It is customary to use 0-indexed players in the whole library.

Reply:
Are you suggesting calling the two players `Player0` and `Player1`?

Reply:
You can use just an `int` (or `Player`, but that is just an int alias)

Reply:
So you are saying substituting every `Player1` with explicitly `Player{0}`? Or just substituting `Player1` with the hardcoded int `0`?

Reply:
Yeah I think Michal is saying you don't need the special names. The players are always integers throughout the whole code base, that is part of the core API. If you do want to keep them I suggest indeed calling them Player0 and Player1 or FirstPlayer and SecondPlayer.

Reply:
OK I feel then perhaps we can go with `Player{0}` and `Player{1}` everywhere instead of `Player1` and `Player2`. This avoids using int literals, and also I think is slightly more readable than `/* player = */ 0` in function calls. I'll amend the code.